1st. Feat(Bento): Implement First Iteration bento design

JosephSanjaya opened the pull request:
## 📱 Description
This PR introduces several new grid components to the Bento home screen, expanding functionality and improving the overall UI structure. The update brings dedicated composables for balances, price tickers, LTC transaction history, and favorite contacts.

## Platform
- [x] Android
- [ ] iOS
- [ ] Games-Unity
- [ ] DevOps (AWS)
- [ ] Website
- [ ] C/Golang

## 🎯 Type of Change
- [x] ✨ New feature
- [x] 🎨 UI/UX improvement
- [ ] 🐛 Bug fix
- [ ] 🔧 Refactoring
- [ ] 📚 Documentation update
- [ ] ⚡ Performance improvement
- [ ] 🧪 Test addition or improvement

## 📋 Changes
- Added `BalanceBentoGrid` composable for LTC and USD balance display.
  - Includes animated visibility toggle (eye icon) with fade transitions.
  - New vector assets for enabled/disabled icons.
- Added `PriceTickerGrid` composable with horizontally swipeable pager to display trading pair prices (e.g., LTC/USD, LTC/BTC).
- Added `TransactionHistoryGrid` composable for displaying LTC transaction history placeholders.
- Added Favorites grid component featuring circular contact avatars and an Add button.
- Updated `BentoMainScreen` layout to integrate all new grid components.
  - Now respects system insets via `statusBarsPadding()` and `navigationBarsPadding()`.
- Enabled edge-to-edge mode in `BentoActivity` for a more immersive UI.
- Updated Gradle packaging rules to exclude duplicate license files.

### 🔗 Related Issues
- Fixes #
- Related to #

## 🧪 Tests Status
- [ ] Tests ran successfully locally
- [ ] Added more tests
- Code coverage percentage of the codebase: __%

## 📸 Screenshots/Videos

| Before | After |
|--------|-------|
| _(previous home layout)_ | _(updated home screen with new grids)_ |

## 🎯 Reviewers
@kcw-grunt, @josikie

JosephSanjaya committed:
chore(design): implement glass container

JosephSanjaya committed:
feat: Implement Balance Bento Grid

This commit introduces the `BalanceBentoGrid` composable, a new component for the home screen's bento layout.

Key changes include:
- A dedicated `BalanceBentoGrid` composable to display LTC and USD balances.
- A visibility toggle (eye icon) to show or hide balance information, enhancing privacy. The state is animated with a fade effect.
- New vector assets for the enabled and disabled eye icons.
- Edge-to-edge display is enabled in `BentoActivity` for a more immersive UI.
- The `BentoMainScreen` is updated to use the new balance grid and to respect system window insets (`statusBarsPadding` and `navigationBarsPadding`).
- Gradle packaging rules are updated to exclude duplicate license files.

JosephSanjaya committed:
feat(ltc): Add TransactionHistoryGrid component

Introduces a new `TransactionHistoryGrid` Composable in the LTC module. This component displays a placeholder for the transaction history on the main screen.

The `BentoMainScreen` is updated to integrate this new component into the grid layout, replacing a generic `BentoHomeGrid` item.

JosephSanjaya committed:
feat(bento): Add PriceTickerGrid composable

Introduces a new `PriceTickerGrid` composable to display cryptocurrency trading pair prices.

This component features a horizontally swipeable pager to cycle through different trading pairs, such as LTC/USD and LTC/BTC, with pagination indicators.

The new grid has been integrated into the `BentoMainScreen` layout.

JosephSanjaya committed:
feat(bento): Add favorite contacts grid

Introduces a new "Favorites" grid component to the main Bento screen.

This new grid displays a list of favorite contacts as circular avatars. It also includes an "Add" button to allow users to add new contacts.

The component is added to the main screen layout, replacing a placeholder grid item.

JosephSanjaya commented:
If you’d like a commit-by-commit review from the past month, this is the first phase of Bento, including the glass → opacity container changes. If this works for you, I’ll create the next PRs step by step. @kcw-grunt if okay with it we can merged this to reduce the changes

kcw-grunt commented on app/src/main/java/com/brainwallet/ui/bento/BentoMainScreen.kt:
```diff
@@ -0,0 +1,230 @@
+            .statusBarsPadding()
+            .navigationBarsPadding()
+    ) {
+        AnimatedLightBleedBackground(
```
Please rename to a ios common name for reference the.  I added this note before .  Dont hardcode layer naming. 

Notes: @JosephSanjaya

I would like to further name this in the comments so that when we compare iOS and Android the code is easily found. .. I dunno on the name but iOS is no better (GridWaveContentView)
Ideas:

Please dont refer it to as Background....View is better because where it is in the view hierarchy is irrelevant.
Here are your choices:

1. AnimatedWaveView
2. ShaderWaveView
3. AnimatedShaderView

cc: @josikie

JosephSanjaya replied:
Done renaming to AnimatedShaderView

JosephSanjaya replied:
@kcw-grunt

kcw-grunt reviewed (changes requested):
Note naming function comments in the view

JosephSanjaya committed:
refactor: Rename AnimatedLightBleedBackground to AnimatedShaderView

The composable `AnimatedLightBleedBackground` has been renamed to `AnimatedShaderView` to better reflect its more general purpose as a view that renders animated shaders, rather than being specific to a "light bleed" effect.

This change is applied across the design system module and its usages, including the `BentoMainScreen` and the preview file.

JosephSanjaya commented:
Do you want me to change all new composable with View suffix? @kcw-grunt
